### PR TITLE
MINOR: Increase max.poll time for streams consumers

### PR DESCRIPTION
**Pull request by enothereska**

### CODE REVIEW

**Comment by enothereska** [comment]
@mjsax @ijuma @dguy @guozhangwang thanks. This should also go into 0.10.2 bug fix release.

File: streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java [hunk 1]
```diff
@@ -401,7 +401,13 @@
...
         tempConsumerDefaultOverrides.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
         tempConsumerDefaultOverrides.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "false");
-        tempConsumerDefaultOverrides.put(ConsumerConfig.MAX_POLL_INTERVAL_MS_CONFIG, "300000");
+        // MAX_POLL_INTERVAL_MS_CONFIG needs to be large for streams to handle cases when
+        // streams is recovering data from state stores. We may set it to Integer.MAX_VALUE since
+        // the streams code itself catches most exceptions and acts accordingly without needing
+        // this timeout. Note however that deadlocks are not detected (by definition) so we
+        // are losing the ability to detect them by setting this value to large. Hopefully
+        // deadlocks happen very rarely or never.
+        tempConsumerDefaultOverrides.put(ConsumerConfig.MAX_POLL_INTERVAL_MS_CONFIG, Integer.toString(Integer.MAX_VALUE));
```
Comment (dguy) [inline streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java:410]: This doesn't stop anyone from overriding this to something else. Should we enforce it or still let users control it?

Comment (mjsax) [reply streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java:410]: I think that's fine. No need to enforce it IMHO.

**Review by mjsax — APPROVED** [review]
LGTM

**Comment by enothereska** [comment]
Yeah, no need to enforce it. If someone changes, presumably they know what they're doing.

**Review by dguy — APPROVED** [review]
LGTM

**Comment by guozhangwang** [comment]
LGTM. Merged to trunk and 0.10.2.

**Pull request closed without merging**